src/chips/nrf51822/gpio.rs

```diff
@@ -0,0 +1,121 @@
+use core::mem;
+use core::ops::{Index, IndexMut};
+use hil;
+
+// Source: https://github.com/hackndev/zinc/tree/master/volatile_cell
+#[derive(Copy, Clone)]
+#[repr(C)]
+pub struct VolatileCell<T> {
```

Comment:
I like this abstraction. We should consider pulling it out and using it more widely